doc/source/index.rst

```diff
@@ -9,84 +9,179 @@ Ray
-|      return 1                                  |      return 1                                      |
-|                                                |                                                    |
-|                                                |                                                    |
-|                                                |  ray.init()                                        |
-|  results = [f() for i in range(4)]             |  results = ray.get([f.remote() for i in range(4)]) |
-+------------------------------------------------+----------------------------------------------------+
-
-To launch a Ray cluster, either privately, on AWS, or on GCP, `follow these instructions <autoscaling.html>`_.
+Install Ray with: ``pip install ray``. For nightly wheels, see the `Installation page <installation.html>`__.
 
 View the `codebase on GitHub`_.
 
 .. _`codebase on GitHub`: https://github.com/ray-project/ray
 
-Ray comes with libraries that accelerate deep learning and reinforcement learning development:
 
-- `Tune`_: Scalable Hyperparameter Search
-- `RLlib`_: Scalable Reinforcement Learning
-- `Distributed Training <distributed_training.html>`__
+Quick Start
+-----------
+
+.. code-block:: python
+
+    ray.init()
+
+    @ray.remote
+    def f(x):
+        return x * x
+
+    futures = [f.remote(i) for i in range(4)]
+    print(ray.get(futures))
+
+To use Ray's actor model:
+
+.. code-block:: python
+
+    ray.init()
+
+    @ray.remote
+    class Counter():
+        def __init__(self):
+            self.n = 0
+
+        def inc(self):
+            self.n += 1
+
+        def read(self):
+            return self.n
+
+    counters = [Counter.remote() for i in range(4)]
+    [c.increment.remote() for c in counters]
+    futures = [c.read.remote() for c in counters]
+    print(ray.get(futures))
+
+
+Ray programs can run on a single machine, and can also seamlessly scale to large clusters. To execute the above Ray script in the cloud, just download `this configuration file <https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-full.yaml>`__, and run:
+
+``ray submit [CLUSTER.YAML] example.py --start``
+
+See more details in the `Cluster Launch page <autoscaling.html>`_.
+
+Tune Quick Start
+----------------
+
+`Tune`_ is a scalable framework for hyperparameter search with a focus on deep learning and deep reinforcement learning.
+
+.. code-block:: python
+
+   import torch.optim as optim
+   from ray import tune
+   from ray.tune.examples.mnist_pytorch import get_data_loaders, Net, train, test
+
+
+   def train_mnist(config):
+       train_loader, test_loader = get_data_loaders()
+       model = Net(config)
+       optimizer = optim.SGD(model.parameters(), lr=config["lr"])
+       for i in range(10):
+           train(model, optimizer, train_loader)
+           acc = test(model, test_loader)
+           tune.track.log(mean_accuracy=acc)
+
+
+   analysis = tune.run(
```

Comment:
This script failed for me with

```
(pid=91873) 2019-07-30 22:01:34,744	ERROR function_runner.py:98 -- Runner Thread raised error.
(pid=91873) Traceback (most recent call last):
(pid=91873)   File "/Users/rkn/Workspace/ray/python/ray/tune/function_runner.py", line 92, in run
(pid=91873)     self._entrypoint()
(pid=91873)   File "/Users/rkn/Workspace/ray/python/ray/tune/function_runner.py", line 140, in entrypoint
(pid=91873)     return self._trainable_func(config, self._status_reporter)
(pid=91873)   File "/Users/rkn/Workspace/ray/python/ray/tune/function_runner.py", line 266, in _trainable_func
(pid=91873)     output = train_func(config)
(pid=91873)   File "<ipython-input-1-fada79183d81>", line 11, in train_mnist
(pid=91873) TypeError: train() missing 1 required positional argument: 'device'
...
...
...
~/Workspace/ray/python/ray/tune/tune.py in run(run_or_experiment, name, stop, config, resources_per_trial, num_samples, local_dir, upload_dir, trial_name_creator, loggers, sync_to_cloud, sync_to_driver, checkpoint_freq, checkpoint_at_end, keep_checkpoints_num, checkpoint_score_attr, global_checkpoint_period, export_formats, max_failures, restore, search_alg, scheduler, with_server, server_port, verbose, resume, queue_trials, reuse_actors, trial_executor, raise_on_failed_trial, return_trials, ray_auto_init, sync_function)
    260     if errored_trials:
    261         if raise_on_failed_trial:
--> 262             raise TuneError("Trials did not complete", errored_trials)
    263         else:
    264             logger.error("Trials did not complete: %s", errored_trials)

TuneError: ('Trials did not complete', [train_mnist_0_lr=0.001, train_mnist_1_lr=0.01, train_mnist_2_lr=0.1])
```

The error message continues on, so I may not have copied the relevant part.